src/main/java/de/tum/in/www1/artemis/service/metis/conversation/GroupChatService.java

```diff
@@ -0,0 +1,120 @@
+        this.groupChatRepository = groupChatRepository;
+    }
+
+    /**
+     * Gets a group chat by its id from the database or throws an exception if it does not exist
+     *
+     * @param groupChatId the id of the group chat
+     * @return the group chat
+     */
+    public GroupChat getGroupChatOrThrow(Long groupChatId) {
+        return groupChatRepository.findById(groupChatId)
+                .orElseThrow(() -> new BadRequestAlertException("GroupChat with id " + groupChatId + " does not exist", GROUP_CHAT_ENTITY_NAME, "idnotfound"));
+    }
+
+    /**
+     * Creates a new group chat
+     *
+     * @param course          the course the group chat is created in
+     * @param startingMembers the users that are initially part of the group chat
+     * @return the newly created group chat
+     */
+    public GroupChat startGroupChat(Course course, Set<User> startingMembers) {
+        var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
+        var groupChat = new GroupChat();
+        groupChat.setCourse(course);
+        groupChat.setCreator(requestingUser);
+        var savedGroupChat = groupChatRepository.save(groupChat);
+        var participantsToCreate = startingMembers.stream().map(user -> createChatParticipant(user, groupChat)).toList();
+        conversationParticipantRepository.saveAll(participantsToCreate);
+        return savedGroupChat;
+    }
+
+    @NotNull
+    private ConversationParticipant createChatParticipant(User user, GroupChat groupChat) {
+        var participant = new ConversationParticipant();
+        participant.setUser(user);
+        participant.setConversation(groupChat);
+        // makes no sense for group chats
+        participant.setIsAdmin(false);
+        participant.setIsFavorite(false);
+        participant.setIsHidden(false);
+        return participant;
+    }
+
+    /**
+     * Updates the group chat with the given id
+     *
+     * @param groupChatId  the id of the group chat to update
+     * @param groupChatDTO the DTO containing the new group chat data
+     * @return the updated group chat
+     */
+    public GroupChat updateGroupChat(Long groupChatId, GroupChatDTO groupChatDTO) {
+        var groupChat = getGroupChatOrThrow(groupChatId);
+        if (groupChatDTO.getName() != null && !groupChatDTO.getName().equals(groupChat.getName())) {
+            groupChat.setName(groupChatDTO.getName().trim().isBlank() ? "" : groupChatDTO.getName().trim());
+        }
+        this.groupChatIsValidOrThrow(groupChat);
+        var updatedGroupChat = groupChatRepository.save(groupChat);
+        conversationService.notifyAllConversationMembersAboutUpdate(updatedGroupChat);
+        return updatedGroupChat;
+    }
+
+    /**
+     * Checks if the group chat is valid and throws an exception if it is not
+     *
+     * @param groupChat the group chat to check
+     */
+    public void groupChatIsValidOrThrow(GroupChat groupChat) {
```

Comment:
Same as above.

Reply:
Done in 37585c59926919519091389f87bcd31d8c84e40c